### PR TITLE
Draft for issue #1012 Show dependencies

### PR DESCRIPTION
**Pull request by fzanollo**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Are you working against an Issue?

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/1013)

### CODE REVIEW

File: doc/specs/#1012 - Show dependencies.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Solution Design
+The Windows Package Manager should be able to show package dependency information for each of the four different types of dependencies declared in the [v1.0 manifest schemas](https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/manifests/v1.0.0/).
+
+* Windows Features
+* Windows Libraries
+* Package Dependencies (same source)
+* External Dependencies
+
+This information will be shown in the commands: show, install and import. Adding to the information already showed.
```
Comment (JohnMcPMS) [inline doc/specs/#1012 - Show dependencies.md:23]: I think that there is at least potential to need work in the following commands:
- install
- show
- upgrade
- uninstall
- validate
- import

In addition to enumerating which commands will begin to acknowledge dependencies, please have a section for each command and detail what will be output.  As an example:

### install
The install command will enumerate the dependencies of the package version being installed as follows:
```
> winget install notepad++
Found Notepad++ [Notepad++.Notepad++]
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
- Windows Feature: Hyper-V
- Package: Microsot.WindowsTerminal
Downloading https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
```

File: src/AppInstallerCLICore/ExecutionReporter.cpp [hunk 2]
```diff
@@ -11,6 +11,7 @@ namespace AppInstaller::CLI::Execution
 
     const Sequence& HelpCommandEmphasis = TextFormat::Foreground::Bright;
     const Sequence& HelpArgumentEmphasis = TextFormat::Foreground::Bright;
+    const Sequence& ManifestKeyEmphasis = TextFormat::Foreground::Bright;
```
Comment (JohnMcPMS) [inline src/AppInstallerCLICore/ExecutionReporter.cpp:14]: Please have the spec PR be just the spec; no code changes.

File: doc/specs/#1012 - Show dependencies.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+last updated: 2021-05-28
+issue id: 1012
+---
+
+# Show dependencies
+
+For [#1012](https://github.com/microsoft/winget-cli/issues/1012)
+
+## Abstract
+Several packages require other packages as dependencies. The Windows Package Manager should be able to support declared dependencies and inform the user about any required dependencies.
```
Comment (JohnMcPMS) [inline doc/specs/#1012 - Show dependencies.md:13]: I would suggest explicitly stating that this is the first step towards full support for dependencies to make it clear that we do not intend to stop at simply informing the user.

**Review by JohnMcPMS — CHANGES_REQUESTED** [review]

File: doc/specs/#1012 - Show dependencies.md (hunk 1 above)
Comment (JohnMcPMS) [reply doc/specs/#1012 - Show dependencies.md:23]: Although uninstall might be a long term goal and could probably be skipped for now as we don't even have the information to inform the user.

**Comment by fzanollo** [comment]
Hi everyone, it wasn't my intention to add code to the PR. 
I think the most clean way is to close this one and start again. Sorry!

**Pull request closed without merging**